quickstart/src/main/java/com/microsoft/azure/kusto/quickstart/KustoSampleApp.java

```diff
@@ -139,75 +138,84 @@ private static void loadConfigs(String configFileName) {
             tableSchema = (String) configs.get("tableSchema");
             kustoUrl = (String) configs.get("kustoUri");
             ingestUrl = (String) configs.get("ingestUri");
             dataToIngest = (List<Map<String, String>>) configs.get("data");
-            alterTable = Boolean.parseBoolean((String) configs.get("alterTable"));
-            queryData = Boolean.parseBoolean((String) configs.get("queryData"));
-            ingestData = Boolean.parseBoolean((String) configs.get("ingestData"));
+            shouldAlterTable = Boolean.parseBoolean((String) configs.get("alterTable"));
+            shouldQueryData = Boolean.parseBoolean((String) configs.get("queryData"));
+            shouldIngestData = Boolean.parseBoolean((String) configs.get("ingestData"));
             if (StringUtils.isBlank(databaseName) || StringUtils.isBlank(tableName) || StringUtils.isBlank(tableSchema)
                     || StringUtils.isBlank(kustoUrl) || StringUtils.isBlank(ingestUrl)
                     || StringUtils.isBlank((String) configs.get("useExistingTable")) || dataToIngest.isEmpty()) {
-                die(String.format("%s is missing required fields", CONFIG_FILE_NAME));
+                die(String.format("File '%s' is missing required fields", configFileName));
             }
         } catch (IOException e) {
-            die("Couldn't read config file", e);
+            die(String.format("Couldn't read config file from file '%s'", configFileName), e);
         }
     }
 
-    private static ConnectionStringBuilder generateConnectionString(String endpointUrl, String authenticationMode) {
+    private static ConnectionStringBuilder generateConnectionString(String clusterUrl, String authenticationMode) {
+        // Learn More: For additional information on how to authorize users and apps in Kusto, see: https://docs.microsoft.com/azure/data-explorer/manage-database-permissions
         ConnectionStringBuilder csb = null;
         switch (authenticationMode) {
             case "UserPrompt":
-                csb = ConnectionStringBuilder.createWithUserPrompt(endpointUrl);
+                // Prompt user for credentials
+                csb = ConnectionStringBuilder.createWithUserPrompt(clusterUrl);
                 break;
             case "ManagedIdentity":
-                // Connect using the system or user provided managed identity (Azure service only)
-                // TODO - Config (Optional): Managed identity client id if you are using a User Assigned Managed Id
-                String clientId = System.getenv("MANAGED_IDENTITY_CLIENT_ID");
-                if (StringUtils.isBlank(clientId)) {
-                    csb = ConnectionStringBuilder.createWithAadManagedIdentity(endpointUrl);
-                } else {
-                    csb = ConnectionStringBuilder.createWithAadManagedIdentity(endpointUrl, clientId);
-                }
+                csb = createManagedIdentityConnectionString(clusterUrl);

```

Comment:
Not sure how this helps.
You just add more and more functions the end user needs to swift through to get to the good stuff

Reply:
Just trying to make it easier for the beginner user to read and follow.

In the existing Python implementation, **each** is its own method, even if just 1 line (e.g. `create_interactive_auth_connection_string`).
![image](https://user-images.githubusercontent.com/30245027/146675744-9e21b9ce-a1b9-402f-b030-e67057fcc5e3.png)